reana/reana_benchmark/monitor.py

```diff
@@ -0,0 +1,192 @@
+        return {
+            self.name: result,
+        }
+
+
+class WorkflowStatuses(BaseMetric):
+    """Count number of workflows statuses directly from DB."""
+
+    name = "workflow_statuses"
+
+    def _collect(self, parameters: Dict) -> Any:
+        workflow_prefix = parameters.get("workflow")
+
+        if not workflow_prefix:
+            logger.warning(
+                f"{self.name} metrics cannot find workflow parameter. Metric will not be collected."
+            )
+            return {}
+
+        cmd = [
+            "kubectl",
+            "exec",
+            "deployment/reana-db",
+            "--",
+            "psql",
+            "-U",
+            "reana",
+            "-c",
+            f"SELECT status,COUNT(*) FROM __reana.workflow WHERE name LIKE '{workflow_prefix}-%' GROUP BY status;",
+        ]
+        output = subprocess.check_output(cmd).decode("ascii")
+        result = {}
+
+        rows = output.splitlines()[2:-2]
+
+        for row in rows:
+            status, count = row.split("|")[0].strip(), int(row.split("|")[1].strip())
+            result[status] = count
+
+        return result
+
+
+class NumberOfDBConnectionsMetric(BaseMetric):
+    """Count number of server processes in REANA DB."""
+
+    name = "db_connections_number"
+
+    def _collect(self, parameters: Dict) -> Any:
+        cmd = [
+            "kubectl",
+            "exec",
+            "deployment/reana-db",
+            "--",
+            "psql",
+            "-U",
+            "reana",
+            "-c",
+            "SELECT COUNT(*) FROM pg_stat_activity;",
+        ]
+        output = subprocess.check_output(cmd).decode("ascii")
+        result = int(output.splitlines()[2].strip())
+        return result
+
+
+class WorkflowPodsMetric(BaseMetric):
+    """Count number of job and batch jobs in different phases."""
+
+    name = "workflows_pods_status"
+
+    @staticmethod
+    def _filter(pods: List[Dict], name_contains: str) -> Generator[Dict, None, None]:
+        for pod in pods:
+            name = pod.get("metadata", {}).get("name", "")
+            if name_contains in name:
+                yield pod
+
+    @staticmethod
+    def _count(pods: List[Dict], name_contains: str) -> Dict[str, int]:
+        statistics = defaultdict(lambda: 0)
+        for pod in WorkflowPodsMetric._filter(pods, name_contains):
+            phase = pod.get("status", {}).get("phase")
+            statistics[phase] += 1
+        return dict(statistics)
+
+    def _collect(self, parameters: Dict) -> Any:
+        kubectl_cmd = ("kubectl", "get", "pods", "-o", "json")
+        output = subprocess.check_output(kubectl_cmd)
+        pods = json.loads(output).get("items", [])
+
+        result = {
+            "batch_pods": self._count(pods, "run-batch"),
+            "job_pods": self._count(pods, "run-job"),
+        }
+
+        return result
+
+
+METRICS = [
+    NumberOfDBConnectionsMetric(),
+    WorkflowPodsMetric(),
+    WorkflowStatuses(),
```

Comment:
Minor: for consistency we could add `Metric` to the end. Also, I think it would be a bit more clear to name it `WorkflowDBStatusesMetric` and accordingly `name = "workflow_db_statuses"`. I think it would be more in line with `workflows_pods_status`

Reply:
Done